src/astro/sql/operators/transform.py

```diff
@@ -0,0 +1,397 @@
+        self.output_table: Table = self.op_kwargs.pop("output_table", Table())
+        self.handler = self.op_kwargs.pop("handler", handler)
+        self.conn_id = self.op_kwargs.pop("conn_id", conn_id)
+        self.sql = sql
+        self.parameters = parameters or {}
+        self.database = (self.op_kwargs.pop("database", database),)
+        self.schema = (self.op_kwargs.pop("schema", schema),)
+        self.raw_sql = raw_sql
+        super().__init__(
+            **kwargs,
+        )
+
+    def execute(self, context: Dict):
+        first_table = find_first_table(
+            op_args=self.op_args,  # type: ignore
+            op_kwargs=self.op_kwargs,
+            python_callable=self.python_callable,
+            parameters=self.parameters or {},  # type: ignore
+        )
+        if first_table:
+            self.conn_id = self.conn_id or first_table.conn_id  # type: ignore
+            self.database = self.database or first_table.metadata.database  # type: ignore
+            self.schema = self.schema or first_table.metadata.schema  # type: ignore
+        else:
+            if not self.conn_id:
+                raise ValueError("You need to provide a table or a connection id")
+        self.database_impl = create_database(self.conn_id)
+
+        # Find and load dataframes from op_arg and op_kwarg into Table
+        self.create_output_table_if_needed()
+        self.op_args = load_op_arg_dataframes_into_sql(
+            conn_id=self.conn_id,
+            op_args=self.op_args,  # type: ignore
+            target_table=self.output_table.create_similar_table(),
+        )
+        self.op_kwargs = load_op_kwarg_dataframes_into_sql(
+            conn_id=self.conn_id,
+            op_kwargs=self.op_kwargs,
+            target_table=self.output_table.create_similar_table(),
+        )
+
+        # Get SQL from function and render templates in the SQL String
+        self.read_sql_from_function()
+        self.move_function_params_into_sql_params(context)
+        self.translate_jinja_to_sqlalchemy_template(context)
+        # if there is no SQL to run we raise an error
+        if self.sql == "" or not self.sql:
+            raise AirflowException("There's no SQL to run")
+
+        if self.raw_sql:
+            result = self.database_impl.run_sql(
+                sql_statement=self.sql, parameters=self.parameters
+            )
+            if self.handler:
+                return self.handler(result)
+            else:
+                return None
+        else:
+            self.database_impl.create_schema_if_needed(
+                self.output_table.metadata.schema
+            )
+            self.database_impl.drop_table(self.output_table)
+            self.database_impl.create_table_from_select_statement(
+                statement=self.sql,
+                target_table=self.output_table,
+                parameters=self.parameters,
+            )
+            return self.output_table
```

Comment:
We could have a separated PR to split the `raw_sql` operator from the `transfrom`, what do you think @dimberman ?